src/GafferImage/Merge.cpp

```diff
@@ -199,7 +199,7 @@ void Merge::hashChannelData( const GafferImage::ImagePlug *output, const Gaffer:
 			continue;
 		}
 
-		IECore::ConstStringVectorDataPtr channelNamesData = (*it)->channelNamesPlug()->getValue();
+		IECore::ConstStringVectorDataPtr channelNamesData = (*it)->channelNames();
```

Comment:
It seems unfortunate that the context construction is inside the loop here. Can we at least share a single GlobalContext with the dataWindow computation below?